src/ASM/LR/ASMLRSpline.C

```diff
@@ -447,6 +447,25 @@ std::pair<size_t,double> ASMLRSpline::findClosestNode (const Vec3& X) const
 }
 
 
+Vec3 ASMLRSpline::getElementCenter (int iel) const
+{
+#ifdef INDEX_CHECK
+  if (iel < 1 || iel > geo->nElements())
+  {
+    std::cerr <<" *** ASMLRSpline::getElementCenter: Element index "<< iel
+              <<" out of range [1,"<< geo->nElements() <<"]."<< std::endl;
+    return Vec3();
+  }
+#endif
+
+  double u0[3] = { 0.0, 0.0, 0.0 };
+  LR::Element* elm = geo->getElement(iel-1);
+  for (unsigned char d = 0; d < ndim; d++)
+    u0[d] = 0.5*(elm->getParmin(d) + elm->getParmax(d));
+  return this->point(u0[0],u0[1],u0[2],iel-1);
```

Comment:
same here, this won't work for ASMu2Dnurbs.

Reply:
replaced by evalPoint()